core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -100,7 +102,18 @@ private[spark] object JsonProtocol {
         executorMetricsUpdateToJson(metricsUpdate)
       case blockUpdate: SparkListenerBlockUpdated =>
         blockUpdateToJson(blockUpdate)
-      case _ => parse(mapper.writeValueAsString(event))
+      case _ =>
+        // Use piped streams to avoid extra memory consumption
+        val outputStream = new PipedOutputStream()
+        val inputStream = new PipedInputStream(outputStream)
+        try {
+          mapper.writeValue(outputStream, event)
```

Comment:
Wait wait .. does this lazily work for sure? Can we add a test (or manual test in the PR description) that reads some more data (maybe more then the buffer size in that pipe)?

Reply:
I was actually hoping for a test to fail, but none did (the test suite has a bunch of very specific stuff). This code will likely block forever if the block size is larger. Going to add a test